File: lab01. комплексные числа.cpp
#include <iostream>
#include <complex>
#include <cmath>

// Функция для форматирования комплексного числа в строку вида "a + bi"
std::string complexToString(const std::complex<double>& c) {
    std::ostringstream oss;
    oss << c.real();
    if (c.imag() >= 0) {
        oss << " + " << c.imag() << "i";
    } else {
        oss << " - " << -c.imag() << "i";
    }
    return oss.str();
}

int main() {
    // Определение комплексных чисел
    std::complex<double> z1(-1, 1);
    std::complex<double> z2(-3, -1);
    std::complex<double> z3(1, 2);

    // Сумма
    std::complex<double> sum = z1 + z2;
    std::cout << "Сумма: " << complexToString(sum) << std::endl;

    // Разность
    std::complex<double> diff = z1 - z2;
    std::cout << "Разность: " << complexToString(diff) << std::endl;

    // Произведение
    std::complex<double> prod = z1 * z2;
    std::cout << "Произведение: " << complexToString(prod) << std::endl;

    // Частное
    std::complex<double> quot = z1 / z2;
    std::cout << "Частное: " << complexToString(quot) << std::endl;

    // Четвертая степень z3
    std::complex<double> z3_pow4 = std::pow(z3, 4);
    std::cout << "z3 в четвертой степени: " << complexToString(z3_pow4) << std::endl;

    // Корень третьей степени из z3
    std::complex<double> z3_root3 = std::pow(z3, 1.0/3);
    std::cout << "Корень третьей степени из z3: " << complexToString(z3_root3) << std::endl;

    return 0;
}

File: lab01. комплексные числа.py
import cmath

# Функция для форматирования комплексного числа в строку вида "a + bi"
def complex_to_string(c):
    real_part = c.real
    imag_part = c.imag
    if imag_part >= 0:
        return f"{real_part} + {imag_part}i"
    else:
        return f"{real_part} - {-imag_part}i"

# Определение комплексных чисел
z1 = complex(-1, 1)
z2 = complex(-3, -1)
z3 = complex(1, 2)

# Сумма
sum_z = z1 + z2
print(f"Сумма: {complex_to_string(sum_z)}")

# Разность
diff_z = z1 - z2
print(f"Разность: {complex_to_string(diff_z)}")

# Произведение
prod_z = z1 * z2
print(f"Произведение: {complex_to_string(prod_z)}")

# Частное
quot_z = z1 / z2
print(f"Частное: {complex_to_string(quot_z)}")

# Четвертая степень z3
z3_pow4 = z3 ** 4
print(f"z3 в четвертой степени: {complex_to_string(z3_pow4)}")

# Корень третьей степени из z3
z3_root3 = z3 ** (1/3)
print(f"Корень третьей степени из z3: {complex_to_string(z3_root3)}")
